controllers/cluster_create.go

```diff
@@ -1248,3 +1238,37 @@ func (r *ClusterReconciler) removeDanglingPVCs(ctx context.Context, cluster *api
 
 	return nil
 }
+
+func (r *ClusterReconciler) createPVC(
+	ctx context.Context,
+	ownedBy *apiv1.Cluster,
+	storageConfiguration apiv1.StorageConfiguration,
+	name string,
```

Comment:
instead of passing the name of the PVC, perhaps we could pass the suffix to be used for this PVC. This would be empty for the data directory, and `wal` for the WAL archive.